core-bundle/src/Twig/Loader/ContaoFilesystemLoader.php

```diff
@@ -104,6 +104,11 @@ public function addPath(string $path, string $namespace = 'Contao', bool $trackT
             throw new LoaderError(sprintf('Tried to register an invalid Contao namespace "%s".', $namespace));
         }
 
+        // Skip execution if the given path and namespace was already registered
+        if (\in_array(rtrim($path, '/\\'), $this->paths[$namespace] ?? [], true)) {
```

Comment:
The `rtrim` is needed, because the parent implementation does the same when adding paths. Another reason to drop it and implement everything directly... :see_no_evil: